src/model/ve_bus_state.rs

```diff
@@ -0,0 +1,105 @@
+use crate::bit_reader::BitReader;
+use crate::err::*;
+use num_enum::TryFromPrimitive;
+
+use super::mode::Mode;
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd)]
+pub struct VeBusState {
+    pub mode: Mode,
+    pub error: u8,
+    pub battery_voltage_v: f32,
+    pub battery_current_a: f32,
+    pub ac_in_state: AcInState,
+    pub ac_in_power_w: f32,
+    pub ac_out_power_w: f32,
+    pub alarm: AlarmNotification,
+    pub battery_temperature_c: f32,
+    pub soc_percent: f32,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, TryFromPrimitive)]
+#[repr(u8)]
+pub enum AcInState {
+    AcIn1 = 0,
+    AcIn2 = 1,
+    NotConnected = 2,
+    Unknown = 3,
+}
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Copy, Clone, PartialEq, PartialOrd, TryFromPrimitive)]
+#[repr(u8)]
+pub enum AlarmNotification {
+    NoAlarm = 0,
+    Warning = 1,
+    Alarm = 2,
```

Comment:
Consider adding the N/A value 3 to the enum, although I haven't done that for other enums I probably should.

```rust
NotApplicable = 3
```